indy_node/test/auth_rule/test_auth_rule_transaction.py

```diff
@@ -116,3 +120,33 @@ def test_reqnack_auth_rule_add_transaction_with_wrong_format(looper,
     e.match("Transaction for change authentication "
             "rule for {}={} must not contain field {}".
             format(AUTH_ACTION, ADD_PREFIX, OLD_VALUE))
+
+
+@pytest.mark.parametrize("off_ledger_signature", [True, False, None])
+def test_auth_rule_state_format(
+    looper, sdk_pool_handle, sdk_wallet_trustee, txnPoolNodeSet, off_ledger_signature
+):
+    auth_action = ADD_PREFIX
+    auth_type = NYM
+    field = ROLE
+    new_value = ENDORSER
+    constraint = {CONSTRAINT_ID: ConstraintsEnum.ROLE_CONSTRAINT_ID,
+                  ROLE: "*",
+                  SIG_COUNT: 1,
+                  NEED_TO_BE_OWNER: False,
+                  METADATA: {}}
+    if off_ledger_signature is not None:
+        constraint[OFF_LEDGER_SIGNATURE] = off_ledger_signature
```

Comment:
Since SDK doesn't send OFF_LEDGER_SIGNATURE in request if it's sent to False, I think we need to expect it in `constraint` only if off_ledger_signature is True.